ui/app/AppLayouts/Profile/views/wallet/NetworksView.qml

```diff
@@ -1,48 +1,103 @@
 import QtQuick 2.13
 
 import shared.status 1.0
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import utils 1.0
 
 import "../../stores"
-import utils 1.0
+import "../../controls"
 
-Column {
+Item {
     id: root
-    anchors.top: parent.top
-    anchors.topMargin: 64
-    anchors.left: parent.left
-    anchors.right: parent.right
+    signal goBack
 
     property WalletStore walletStore
 
-    signal goBack
-
-    ListView {
-        id: layer1List
-        model: walletStore.layer1Networks
-        width: parent.width
-        height: layer1List.childrenRect.height
-        delegate: StatusSettingsLineButton {
-            text: model.chainName
+    StatusFlatButton {
+        id: backButton
+        anchors.top: parent.top
+        anchors.topMargin: 20
+        anchors.left: parent.left
+        anchors.leftMargin: 20
+        icon.name: "arrow-left"
+        icon.height: 13.5
+        icon.width: 17.5
+        text: qsTr("Wallet")
+        onClicked: {
+            root.goBack()
         }
     }
 
-    ListView {
-        id: layer2List
-        model: walletStore.layer2Networks
-        width: parent.width
-        height: layer2List.childrenRect.height
-        delegate: StatusSettingsLineButton {
-            text: model.chainName
+    Column {
+        id: column
+        anchors.topMargin: 30
+        anchors.top: backButton.bottom
+        anchors.leftMargin: 64
+        anchors.left: root.left
+        width: 560
+
+        Row {
+            spacing: 250
+            StatusBaseText {
+                id: titleText
+                text: qsTr("Networks")
+                font.weight: Font.Bold
+                font.pixelSize: 28
+                color: Theme.palette.directColor1
+            }
+
+            StatusButton {
+                id: addCustomNetworkButton
+                type: StatusFlatRoundButton.Type.Primary
+                text: qsTr("Add Custom Network")
+                onClicked: {
+                    root.goBack()
+                }
+            }
+        }
+        
+
+        Item {
+            height: Style.current.bigPadding
+            width: parent.width
+        }
+
+        Repeater {
+            id: layer1List
+            model: walletStore.layer1Networks
+            delegate: NetworkDelegate {
+                network: model
+            }
+        }
+
+        StatusSectionHeadline {
+            text: qsTr("Layer 2")
+            topPadding: Style.current.bigPadding
+            bottomPadding: Style.current.padding
+        }
+
+        Repeater {
+            id: layer2List
+            model: walletStore.layer2Networks
+            delegate: NetworkDelegate {
+                network: model
```

Comment:
this is not needed, model is passed to the delegate already, you can access it the properties in NetworkDelegate  by model.name etc, or even directly name (if I am not mistaken)

Reply:
As it is a different file I believe it is better to specify it no?

Reply:
actually no, that's how delegates usually behave. unless off course if some else used it someone who isn't a list? in that case would be ideal to be able to pass value via properties or aliases from the delegate. Maybe this is preference thing, so I will approve anyway